### ISSUE TITLE
Feature: Upgrade to latest Qt 5.15

### ISSUE THREAD

**Issue by mpogue2** [mpogue2/SquareDesk#543]
We are on 5.9.3, due to the "can't handle a lot of songs in the songTable" problem.
But, from the Qt webpage:
"5.9 LTS, which was released on 31 May 2017, is also still supported – until 31 May 2020"

So, support is going away, and also we're on a 2.5 year old release.  Latest is 5.14 .
We should switch at some point, on all platforms.

**Comment by mpogue2** [comment]
Apple no longer supports OpenGL either, in favor of Metal.
So, we are currently getting this message on Catalina:
"Unable to create basic Accelerated OpenGL renderer.
Core Image is now using the software OpenGL renderer. This will be slow."

**Comment by mpogue2** [comment]
See also #492.

**Comment by mpogue2** [comment]
5.9.3 is officially unsupported now.  I am in the process of upgrading to 5.15 now, AND upgrading to the latest QtCreator, AND upgrading to "QMAKE_MAC_SDK = macosx10.15" (in test123.pro).

So far:
1) There are conflicts with the Homebrew versions of libs.  For some reason, the linker is picking up the Homebrew versions in /usr/local/lib, rather than the system versions.  This appears to be a common problem.  I had to do this:

brew uninstall --ignore-dependencies libjpeg
brew uninstall --ignore-dependencies giflib
brew uninstall --ignore-dependencies libpng

2) With everything upgraded, SquareDesk seems to work fine so far.

3) I have not yet tested for the "can't handle a lot of songs in the songTable" problem yet.

**Comment by mpogue2** [comment]
4. With the latest QtCreator, compilation seems a lot faster than with previous versions.  All 4 CPUs on my laptop are pegged during a clean compile.

**Comment by mpogue2** [comment]
5. For reference, build output is now in:
build-SquareDesk-Desktop_Qt_5_15_0_clang_64bit-Release

**Comment by mpogue2** [comment]
6. THIS is how to get the latest developer tools from Apple. 
```
xcode-select —install
```
7. Also for reference, THIS tells us what we have from Apple:
```
ls /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/
```
8. And, THIS is the magic line to change, in test123.pro, to upgrade to a new kit from Apple (which should be found in that last 'ls' folder:
```
QMAKE_MAC_SDK = macosx10.15
```

This is documented in test123.pro, it's just sometimes hard to remember where.

**Comment by mpogue2** [comment]
#468, the "can't do more than 1000 files" problem does not exist in Qt5.15 .  However, the other two problems noted earlier are still there:
1) all the slider grooves are offset, and 
2) songTable headers go bold when you click inside the songTable

I am inclined to move forward at this point, and fix those 2 problems, rather than remaining on 5.9.3 .  I'm sure there are other bugs fixed in the intervening years of releases that would help us.

**Comment by mpogue2** [comment]
Slider grooves fixed by: 0809d9bcb43c6df6a69e292781899b65f46a6907 .
bold songTable headers fixed by: be35ca6eaafbb272d3ec423d62fd8fbe9df54622 .

OK, I think we're fully up and running now with 5.15, the latest QtCreator, and the latest macossdk.  So, closing this one as Fixed.  @danlyke Please let me know if anything is messed up on Linux.  I am inclined to drop support for Windows at this point, so I don't think there's an issue there.

**Comment by mpogue2** [comment]
The singing call colors needed to move down to match the slider groove moving down.
Fixed by: 21b703284adefbfa55a80b1814988101c314e498 .

**Comment by mpogue2** [comment]
Did it for the Lyrics slider to match: 605ab1c9e2a7e242d95f2a04d2e6d39c225231cd .
